src/system/unwind.c

```diff
@@ -96,9 +96,18 @@ _Unwind_Ptr __gnu_Unwind_Find_exidx(_Unwind_Ptr pc, int* nrec) {
 	return (_Unwind_Ptr)&__exidx_start;
 }
 
+struct {
+	uint32_t[12] pcs;
+	size_t size;
+} trace;
+
 _Unwind_Reason_Code trace_fn(_Unwind_Context* unwind_ctx, void* d) {
 	uint32_t pc = _Unwind_GetIP(unwind_ctx);
 	fprintf(stderr, "\t%p\n", (void*)pc);
+	if (trace.size < sizeof(trace.pcs) / sizeof(trace.pcs[0]))
+		trace.pcs[trace.size++] = pc;
+	else
+		; // TODO: handle this
```

Comment:
Can you please add curly brackets to the if and else cases. If you want suggestions for how to do the else case please reach out, I'm happy to help brainstorm the desired behavior here. I personally think that trying to get dynamically sized trace will be more useful or if you are to hard code in the cap for the length of the array then you shouldn't even need a else case.

Reply:
Added curly braces, also added an else case